src/main/java/org/zalando/nakadi/validation/SchemaEvolutionService.java

```diff
@@ -77,19 +86,42 @@ public EventType evolve(final EventType original, final EventTypeBase eventType)
 
         if (incompatibility.isPresent()) {
             throw new InvalidEventTypeException(incompatibility.get().getReason());
-        } else {
-            final List<SchemaChange> changes = schemaDiff.collectChanges(schema(original), schema(eventType));
+        }
+
+        final List<SchemaChange> changes = schemaDiff.collectChanges(schema(original), schema(eventType));
 
-            final Version.Level changeLevel = semanticOfChange(changes);
+        final Version.Level changeLevel = semanticOfChange(changes);
 
-            if (eventType.getCompatibilityMode() == CompatibilityMode.COMPATIBLE && changeLevel == MAJOR) {
+        validateCompatibleModeChanges(original, changes, changeLevel);
+        validateCompatibilityModeMigration(original, eventType, changes);
+
+        return this.bumpVersion(original, eventType, changeLevel);
+    }
+
+    private void validateCompatibilityModeMigration(final EventType original, final EventTypeBase eventType,
+                                                    final List<SchemaChange> changes) throws InvalidEventTypeException {
+        if (original.getCompatibilityMode() == CompatibilityMode.FIXED
+            && eventType.getCompatibilityMode() == CompatibilityMode.COMPATIBLE) {
+            final List<SchemaChange.Type> changesTypes = changes.stream().map(SchemaChange::getType)
+                    .collect(Collectors.toList());
+            if (!FIXED_TO_COMPATIBLE_ALLOWED_CHANGES.containsAll(changesTypes)) {
```

Comment:
It is up to developer how to do it this check, but here we can avoid creating a list by using allMatch.